CI: ci-integration - TestLBServiceResolver

## CI failure

```
--- FAIL: TestLBServiceResolver (0.07s)

    logger.go:256: time=2025-12-04T18:28:01.569Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:385 msg="Starting hive"
    logger.go:256: time=2025-12-04T18:28:01.570Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:389 msg="Started hive" duration=488.664µs
    resolver_test.go:148: 
        	Error Trace:	/home/runner/work/cilium/cilium/pkg/dial/resolver_test.go:148
        	            				/home/runner/work/cilium/cilium/pkg/dial/resolver_test.go:51
        	Error:      	Should be true
        	Test:       	TestLBServiceResolver
        	Messages:   	The store should have started
    logger.go:256: time=2025-12-04T18:28:01.624Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:401 msg="Stopping hive"
    logger.go:256: time=2025-12-04T18:28:01.625Z level=INFO source=/home/runner/work/cilium/cilium/vendor/github.com/cilium/hive/hive.go:405 msg="Stopped hive" duration=585.594µs
```


```
go test -c ./pkg/dial && stress  ./dial.test
...
10s: 702 runs so far, 5 failures (0.71%), 16 active
```

run: https://github.com/cilium/cilium/actions/runs/19939297068

Related commit that introduced this test: https://github.com/cilium/cilium/commit/0e0242ec02e

--- a/pkg/dial/resolver_test.go
+++ b/pkg/dial/resolver_test.go
@@ -107,19 +107,20 @@ func testResolver(t *testing.T, cells ...cell.Cell) {
 
 		cell.Invoke(func(cl_ *k8sClient.FakeClientset, resolver_ dial.Resolver) {
 			cl = cl_
+
+			// Add a reactor to check that services are being watched.
+			cl.SlimFakeClientset.PrependReactor("list", "services",
+				func(action k8stest.Action) (handled bool, ret runtime.Object, err error) {
+					started.Store(true)
+					return false, nil, nil
+				},
+			)
 			resolver = resolver_
 		}),
 		cell.Group(cells...))
 
 	require.NoError(t, h.Start(tlog, ctx))
 	t.Cleanup(func() { require.NoError(t, h.Stop(tlog, ctx)) })
-
-	cl.SlimFakeClientset.PrependReactor("list", "services",
-		func(action k8stest.Action) (handled bool, ret runtime.Object, err error) {
-			started.Store(true)
-			return false, nil, nil
-		},
-	)
 
 	_, err := cl.Slim().CoreV1().Services("bar").Create(ctx, &slim_corev1.Service{
 		ObjectMeta: slim_metav1.ObjectMeta{Name: "foo", Namespace: "bar"},
@@ -137,7 +138,6 @@ func testResolver(t *testing.T, cells ...cell.Cell) {
 	host, port := resolver.Resolve(ctx, "foo.bar.com", "8080")
 	require.Equal(t, "foo.bar.com", host)
 	require.Equal(t, "8080", port)
-	require.False(t, started.Load(), "The store should not have started")
 
 	require.EventuallyWithT(t, func(c *assert.CollectT) {
 		host, port := resolver.Resolve(ctx, "foo.bar", "8080")

